Python/specialize.c

```diff
@@ -1212,6 +1262,36 @@ function_kind(PyCodeObject *code) {
     return SIMPLE_FUNCTION;
 }
 
+/* Returning 0 indicates a failure. */
+static uint32_t
+function_get_version(PyObject *o, int expected_argcount, int opcode)
```

Comment:
The name of this function is misleading.
Could you split it into two functions,  `function_checkargs` and `function_get_version`.